app/config/FrontendAppConfig.scala

```diff
@@ -26,36 +26,37 @@ import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 @Singleton
 class FrontendAppConfig @Inject() (configuration: Configuration, servicesConfig: ServicesConfig) {
 
-  val host: String    = configuration.get[String]("host")
-  val appName: String = configuration.get[String]("appName")
+  lazy val host: String    = configuration.get[String]("host")
+  lazy val appName: String = configuration.get[String]("appName")
 
-  private val contactHost                  = configuration.get[String]("contact-frontend.host")
-  private val contactFormServiceIdentifier = "income-tax-self-employment-frontend"
+  private lazy val contactHost                  = configuration.get[String]("contact-frontend.host")
+  private lazy val contactFormServiceIdentifier = "income-tax-self-employment-frontend"
 
   def feedbackUrl(implicit request: RequestHeader): String =
     url"$contactHost/contact/beta-feedback?service=$contactFormServiceIdentifier&backUrl=${host + request.uri}".toString
 
-  val loginUrl: String         = configuration.get[String]("urls.login")
-  val loginContinueUrl: String = configuration.get[String]("urls.loginContinue")
-  val signOutUrl: String       = configuration.get[String]("urls.signOut")
+  def loginUrl: String         = configuration.get[String]("urls.login")
```

Comment:
Please correct me if i have overlooked something here, was there a reason we changed it to def?

Reply:
It's so that it can be mocked with `scalamock` - as you can't mock a val with that. So this has been refactored now so that there is a `trait` with a `def` and these can then override with concrete `val` still:
https://scalamock.org/user-guide/faq/#can-i-mock-val--lazy-val